src/test/java/io/orkes/conductor/client/api/MetadataClientTests.java

```diff
@@ -23,37 +22,96 @@
 import io.orkes.conductor.client.MetadataClient;
 import io.orkes.conductor.client.http.OrkesMetadataClient;
 import io.orkes.conductor.client.model.TagObject;
+import io.orkes.conductor.client.model.TagString;
 import io.orkes.conductor.client.util.Commons;
 import io.orkes.conductor.client.util.WorkflowUtil;
 
 public class MetadataClientTests extends ClientTest {
     private final MetadataClient metadataClient;
 
     public MetadataClientTests() {
-        this.metadataClient = super.orkesClients.getMetadataClient();
+        metadataClient = super.orkesClients.getMetadataClient();
     }
 
     @Test
-    @DisplayName("tag a workflows and task")
-    public void tagWorkflowsAndTasks() {
-        registerTask();
-        registerWorkflow();
-        TagObject tagObject = new TagObject();
-        tagObject.setType(TagObject.TypeEnum.METADATA);
-        tagObject.setKey("a");
-        tagObject.setValue("b");
-        ((OrkesMetadataClient) metadataClient).addTaskTag(tagObject, Commons.TASK_NAME);
-        ((OrkesMetadataClient) metadataClient).addWorkflowTag(tagObject, Commons.WORKFLOW_NAME);
-    }
-
-    void registerTask() {
+    void task() {
         TaskDef taskDef = new TaskDef();
         taskDef.setName(Commons.TASK_NAME);
-        this.metadataClient.registerTaskDefs(List.of(taskDef));
+        metadataClient.unregisterTaskDef(Commons.TASK_NAME);
+        metadataClient.registerTaskDefs(List.of(taskDef));
+        metadataClient.updateTaskDef(taskDef);
+        metadataClient.getTaskDef(Commons.TASK_NAME);
     }
 
-    void registerWorkflow() {
+    @Test
+    void workflow() {
         WorkflowDef workflowDef = WorkflowUtil.getWorkflowDef();
+        metadataClient.unregisterWorkflowDef(
+                Commons.WORKFLOW_NAME,
+                Commons.WORKFLOW_VERSION);
         metadataClient.registerWorkflowDef(workflowDef);
+        metadataClient.updateWorkflowDefs(List.of(workflowDef));
+        ((OrkesMetadataClient) metadataClient).updateWorkflowDefs(
```

Comment:
why type cast?

Reply:
The method is available at `Orkes` version of the client, should we update the interface to support it also?

Reply:
Use only the methods from WorkflowClient.